rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,311 @@
+                lp_tokens={addr[1] for addr in token_lp_addresses},  # just the token address
+                str_chain_id=str_chain_id,
+            )
+
+    return (pools,)
+
+
+def ensure_gearbox_tokens_existence(
+        evm_inquirer: 'EvmNodeInquirer',
+        all_pools: list[GearboxPoolData],
+) -> list[GearboxPoolData]:
+    """This function receives data about gearbox pools and ensures that lp tokens and pool coins
+    exist in rotki's database."""
+    verified_pools = []
+    for pool in all_pools:
+        underlying_token = []
+        # Ensure pool coins exists and underlying tokens in the globaldb.
+        if pool.underlying_token is not None:
+            try:
+                get_or_create_evm_token(
+                    userdb=evm_inquirer.database,
+                    evm_address=pool.underlying_token,
+                    chain_id=evm_inquirer.chain_id,
+                    evm_inquirer=evm_inquirer,
+                    encounter=TokenEncounterInfo(
+                        description='Querying gearbox pools',
+                        should_notify=False,
+                    ),
+                )
+                underlying_token.append(
+                    UnderlyingToken(
+                        address=pool.underlying_token, token_kind=EvmTokenKind.ERC20, weight=ONE,
+                    ),
+                )
+
+            except NotERC20Conformant as e:
+                log.error(
+                    f'Skipping pool {pool} because {pool.underlying_token} is not a '
+                    f'valid ERC20 token. {e}',
+                )
+                continue
+
+        # ensure the lp and farming tokens exist in the db
+        verified_tokens: defaultdict[str, set[str]] = defaultdict(set)
+        for token_type, token_addresses in zip(('pools', 'farm'), (pool.lp_tokens, [pool.farming_pool_token]), strict=False):  # noqa: E501
+            for token_address in token_addresses:
+                try:
+                    token = get_or_create_evm_token(
+                        userdb=evm_inquirer.database,
+                        evm_address=string_to_evm_address(token_address),
+                        chain_id=evm_inquirer.chain_id,
+                        protocol=CPT_GEARBOX,
+                        evm_inquirer=evm_inquirer,
+                        underlying_tokens=underlying_token,
+                        encounter=TokenEncounterInfo(
+                            description='Querying gearbox pools',
+                            should_notify=False,
+                        ),
+                    )
+                    verified_tokens[token_type].add(token.identifier)
+                except NotERC20Conformant as e:
+                    log.error(
+                        f'Skipping pool {pool} because {token_address} is not a '
+                        f'valid ERC20 token. {e}',
+                    )
+                    continue
+
+        pool.farming_pool_token = verified_tokens['farm'].pop()
```

Comment:
why the pop here?

Reply:
To get the farm address as it's still in a set